api/user/index.ts

```diff
@@ -45,17 +45,25 @@ export async function refreshRoleAssignmentForUser(userInfo: UserInfo, req: any)
   delete headers['accept'];
   try {
     const response: AxiosResponse = await http.get(path, { headers });
-    locationInfo = getRoleAssignmentInfo(response.data.roleAssignmentResponse);
-    const roles = getOrganisationRoles(response.data.roleAssignmentResponse);
+    const activeRoleAssignments = getActiveRoleAssignments(response.data.roleAssignmentResponse)
+    locationInfo = getRoleAssignmentInfo(activeRoleAssignments);
+    const roles = getOrganisationRoles(activeRoleAssignments);
     userInfo.roles = userInfo.roles.concat(roles);
     userInfo.roleCategory = getMappedRoleCategory(userInfo.roles, ['LEGAL_OPERATIONS', 'JUDICIAL', 'ADMIN']);
-    req.session.roleAssignmentResponse = response.data.roleAssignmentResponse;
+    req.session.roleAssignmentResponse = activeRoleAssignments;
   } catch (error) {
     console.log(error);
   }
   return locationInfo;
 }
 
+export function getActiveRoleAssignments(roleAssignments: RoleAssignment[]): RoleAssignment[] {
```

Comment:
Instead of Date() hard coded introduce a variable for it and pass it an an argument. This means that you can add more unit tests which are not current date dependant. Please add another 3-4 unit tests with all scenarios. This is quite crucial. hence, asking more test coverage. Thanks